jax/experimental/jax2tf/call_tf.py

```diff
@@ -55,13 +55,15 @@
 zip = util.safe_zip
 
 TfConcreteFunction = Any
+TfVal = jax2tf_internal.TfVal
 
 # The platforms for which to use DLPack to avoid copying (only works on GPU
 # and CPU at the moment, and only for DeviceArray). For CPU we don't need
 # DLPack, if we are careful.
 _DLPACK_PLATFORMS = ("gpu",)
 
-def call_tf(callable_tf: Callable, has_side_effects=True) -> Callable:
+def call_tf(callable_tf: Callable, has_side_effects=True,
+            output_shape_dtype=None) -> Callable:
```

Comment:
can we specify the return type of output_shape_dtype here ? e.g 
output_shape_dtype: Optional[Union[jax.ShapeDtypeStruct, jax.Array]]=None.

It helps catch error if users give illegal data.

Reply:
Unfortunately there is no good type annotation here to put under `Optional`, because if the function returns more than one result then it should have a matching structure. We have this problem elsewhere in JAX, and we have decided to not use typing annotations because those would cause more trouble than help in some cases. The only clearly useful part of the typing annotation is `Optional`, but I think it is not worth writing `Optional[Any]` when the default value is `None`.